File: WaterEffect/VelocityField.cs
public class VelocityField
{
    public float[,] X { get; }
    public float[,] Y { get; }

    public int Size { get; }

    public VelocityField(int size)
    {
        Size = size;
        X = new float[size, size];
        Y = new float[size, size];
    }

    public void Clear()
    {
        System.Array.Clear(X, 0, X.Length);
        System.Array.Clear(Y, 0, Y.Length);
    }
}

File: WaterEffectSwift/Sources/WaterEffectSwift/VelocityField.swift
import Foundation

class VelocityField {
    var x: [[Float]]
    var y: [[Float]]
    let size: Int

    init(size: Int) {
        self.size = size
        self.x = Array(repeating: Array(repeating: 0.0, count: size), count: size)
        self.y = Array(repeating: Array(repeating: 0.0, count: size), count: size)
    }

    func clear() {
        for i in 0..<size {
            for j in 0..<size {
                x[i][j] = 0
                y[i][j] = 0
            }
        }
    }
}
